odo/backends/csv.py

```diff
@@ -140,18 +140,25 @@ class CSV(object):
         If the csv file has a header or not
     encoding : str (default utf-8)
         File encoding
+    dshape: datashape or string representation
+        used specified datashape
     kwargs : other...
         Various choices about dialect
     """
     canonical_extension = 'csv'
 
     def __init__(self, path, has_header=None, encoding='utf-8',
-                 sniff_nbytes=10000, **kwargs):
+                 sniff_nbytes=10000, dshape=None, **kwargs):
         self.path = path
         self._has_header = has_header
         self.encoding = encoding or 'utf-8'
         self._kwargs = kwargs
         self._sniff_nbytes = sniff_nbytes
+        if dshape:
+            if isinstance(dshape, (str, unicode)):
+                dshape = datashape.dshape(dshape)
+            dshape = None if isdimension(dshape.subshape[0][0])  else dshape
```

Comment:
What is this logic for?  Don't you want to test if `isrecord(dshape)` and raise an exception if `False`?

If an invalid dshape is passed in we should raise an exception, not silently swallow it...